docs/fastdds/dds_layer/core/policy/policy.rst

```diff
@@ -1,5 +1,2259 @@
+  :ref:`resourcelimitsqospolicy`, which means that its value must be lower or equal to max_samples_per_instance.
+
+.. note::
+     This QoS Policy concerns to Topic, DataWriter and DataReader entities.
+     :raw-html:`<br />`
+     It cannot be changed after the creation of the entities.
+
+.. _historyqospolicykind:
+
+HistoryQosPolicyKind
+""""""""""""""""""""
+There are two possible values:
+
+* **Keep Last**: The service will only attempt to keep the most recent values of the instance and discard the older
+  ones.
+  The maximum number of samples to keep and deliver is defined by the `depth` of the HistoryQosPolicy, which needs to
+  be consistent with the :ref:`resourcelimitsqospolicy` settings.
+* **Keep All**: The service will attempt to keep all the values of the instance until it can be delivered to all the
+  existing Subscribers.
+  If this option is selected, the depth will not have any effect, so the history is only limited by the values set in
+  :ref:`resourcelimitsqospolicy`.
+  If the limit is reached, the behavior of the system depends on the :ref:`reliabilityqospolicy`, if its kind is
+  BEST_EFFORT the older values will be discarded, but if it is RELIABLE the service blocks the DataWriter until the old
+  values are delivered to all existing Subscribers.
+
+
+Example
+"""""""
+
+C++
+***
+.. literalinclude:: ../../../../../code/DDSCodeTester.cpp
+   :language: c++
+   :dedent: 8
+   :start-after: //DDS_CHANGE_HISTORY_QOS_POLICY
+   :end-before: //!
+
+XML
+***
+.. literalinclude:: /../code/XMLTester.xml
+    :language: xml
+    :start-after: <!-->XML-HISTORY<-->
+    :end-before: <!--><-->
+
+.. _latencybudgetqospolicy:
+
+LatencyBudgetQosPolicy
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. warning::
+    This QoS Policy will be implemented in future releases.
+
+This QoS Policy urgency of the data communication, specifying the maximum acceptable delay from the time the data is
+written until the data is inserted on the cache of the receiver application and notified of the fact.
+That delay by default is set to 0 in order to optimize the internal operations.
+
+List of QoS Policy data members:
+
++--------------------------+-------------------------------+-------------------+
+| Data Member Name         | Type                          | Default Value     |
++==========================+===============================+===================+
+| duration                 | fastrtps::Duration_t          | c_TimeZero        |
++--------------------------+-------------------------------+-------------------+
+
+.. note::
+     This QoS Policy concerns to Topic, DataWriter and DataReader entities.
+     :raw-html:`<br />`
+     It can be changed after the creation of the entities.
+
+.. warning::
+    For DataWriters and DataReaders to match, they must follow the compatibility rule. See :ref:`latencybudget_compatibilityrule`
+    for further details.
+
+.. _latencybudget_compatibilityrule:
+
+Compatibility Rule
+""""""""""""""""""
```

Comment:
Missing an example here

Reply:
This Qos is not implemented and we said that they will not have examples.